pkg/server/restart.go

```diff
@@ -101,6 +103,12 @@ func (c *criContainerdService) recover(ctx context.Context) error {
 		return fmt.Errorf("failed to load images: %v", err)
 	}
 	for _, image := range images {
+		// Checking existence of top-level snapshot for each image being recovered.
```

Comment:
Why not add this logic into `loadImages` function?

Reply:
Because we need call c.client.SnapshotService.

Reply:
Just pass it in. :)

Reply:
I feel like logically this should be part of `loadImages`, we load it only when it is ready.

Reply:
ok